spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingAutoConfiguration.java

```diff
@@ -40,6 +45,7 @@
 @Configuration
 @ConditionalOnBean(ChannelBindingService.class)
 @EnableConfigurationProperties(DefaultPollerProperties.class)
+@AutoConfigureBefore(EndpointAutoConfiguration.class)
```

Comment:
Though I agree that it is better to create the `Channels Endpoint` defined in this configuration class before the `EndpointAutoConfiguration`, I don't see anything specific in this `EndpointAutoConfiguration` class that needs to be done before this configuration class. What am I missing?

Also, does this change anyway impact the way health indicators are aligned in this PR?